### PR TITLE
Update RevealOptions interface for reveal.js

### PR DESCRIPTION
**Pull request by farbodsz**
The options for `reveal.js` seem outdated, and they did not align with [what is currently in the reveal.js API documentation](https://github.com/reveal/revealjs.com/blob/master/src/config.md) so I have updated them accordingly.

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

---

### CODE REVIEW

File: types/reveal/index.d.ts [hunk 1]
```diff
@@ -2,7 +2,8 @@
 // Project: https://github.com/hakimel/reveal.js/
 // Definitions by: robertop87 <https://github.com/robertop87>,
 //                 Nava2 <https://github.com/Nava2>,
-//                 JPtenBerge <https://github.com/JPtenBerge>
+//                 JPtenBerge <https://github.com/JPtenBerge>,
+//                 farbodsz <https://github.com/farbodsz>
```
Comment (domdomegg) [inline types/reveal/index.d.ts:6]: Be aware of what this means - this is not just attribution. But happy to approve if you do want to be a definition owner.

https://github.com/DefinitelyTyped/DefinitelyTyped#definition-owners

**Review by domdomegg — APPROVED** [review]
LGTM. Thanks for these updates!

File: types/reveal/index.d.ts [hunk 2]
```diff
@@ -82,77 +83,117 @@ interface RevealStatic {
...
-    theme?: string | undefined;
+    // https://github.com/reveal/revealjs.com/blob/master/src/code.md
+    highlight?: any;
+
+    // https://github.com/reveal/revealjs.com/blob/master/src/markdown.md
+    markdown?: any;
 
-    // Presentation Size
-    // https://github.com/hakimel/reveal.js/#presentation-size
+    // https://github.com/reveal/revealjs.com/blob/master/src/math.md
```
Comment (domdomegg) [inline types/reveal/index.d.ts:175]: Please bump the version number from 4.1.0 to 4.2 on line 1 for these

File: types/reveal/index.d.ts (hunk 1 above)
Comment (farbodsz) [reply types/reveal/index.d.ts]: Thanks - yes I didn't mean to be added as a definition owner, I should've read that beforehand! Fixed now

**Review by domdomegg — APPROVED** [review]
Thanks for the contribution 🎉

**Comment by farbodsz** [comment]
Ready to merge

**Pull request merged**